### ISSUE TITLE
Cosmetic filter prevents loading certain newsletters

### ISSUE THREAD

**Issue by SEXXES** [uBlockOrigin/uAssets#3430]
please see here:
https://github.com/uBlockOrigin/uBlock-issues/issues/211
(posted originally in the wrong forum)

**Comment by ZaphodBeebblebrox** [comment]
Please use the template, or at the very least provide a link to where your issue occurs. It is nearly impossible to understand what is happening and how to fix it without being able to reproduce it.

**Comment by SEXXES** [comment]
It's only happening in newsletters in my mailbox, that's why I made some screenshots
on the webpage of "dailytalk.ch" is no such problem

IF I disable "cosmetic filter", the newsletter shows as in the 2nd picture…

**Comment by ZaphodBeebblebrox** [comment]
If you open the [logger](https://github.com/gorhill/uBlock/wiki/The-logger) you should be able to see which cosmetic filter is causing your issue and disable it by pasting it in my filters and replacing the `##` with `@#@`.

**Comment by SEXXES** [comment]
already done/tried

hold on, I make a screenshot

**Comment by SEXXES** [comment]
![all filters on](https://user-images.githubusercontent.com/42209961/45258200-45b86e00-b3b4-11e8-83d5-584221c678f0.jpeg)
nothing off; all filters ON, cosmetic filter ON
==> newsletter NOT showing properly (as in image 1 in my original post)

![2 mentioned filters off](https://user-images.githubusercontent.com/42209961/45258206-579a1100-b3b4-11e8-80f9-fec82c39cc3f.jpeg)
the two mentioned filters OFF, cosmetic filter ON
==> newsletter NOT showing properly (as in image 1 in my original post)

![cosmetic filter off](https://user-images.githubusercontent.com/42209961/45258226-8912dc80-b3b4-11e8-81c6-ae2b99ce2fcc.jpeg)
the two mentioned filters ON, cosmetic filter OFF
==> newsletter showing properly (as in image 2 in my original post)

**Comment by okiehsch** [comment]
Ther are no `dom` entries in your loger output with `ll filters ON, cosmetic filter ON` ?

**Comment by SEXXES** [comment]
what means "dom"?

possible, I just disabled the two shown on the first screenshot (Adguard Spyware and EasyPrivacy) but only disabling "Cosmetic Filter" made a real change

**Comment by okiehsch** [comment]
You disabled the two network request filters, they have nothing to do with cosmetic filters.



The color of a row of your logger output hints at how the resource was filtered:

- No color: The request for the resource was allowed to go through because it matched no filter/rule.
- Red: The request for the resource was canceled (--) because of a block filter/rule.
- Green: The request for the resource was allowed to go through (++) because it matched a filter/rule whose purpose is to bypass a matching block filter/rule.
- Yellow:
A DOM element which was blocked by a cosmetic filter; OR
A blocked request for a resource was redirected (<<) to a local replacement resource.

If I go to for example `dailytalk.ch` and type `dom` in the "filter log entries" box I get
![image](https://user-images.githubusercontent.com/16838044/45258529-91b9e180-b3b9-11e8-8d44-28f5255c4c22.png)

**Comment by SEXXES** [comment]
Oh I see… yes but:

it only happens when I want to read the newsletter (e-mail) message from dailytalk.ch

about the colors: yes I know, but however: when I disable cosmetic filter, the newsletter is shown properly

**Comment by okiehsch** [comment]
> it only happens when I want to read the newsletter (e-mail) message from dailytalk.ch

I understand that but if you open your newsletter via `https://mail.protonmail.com` with cosmetic filters **enabled**, are there any yellow `dom` elements in your logger output?

**Comment by SEXXES** [comment]
no, no colors, everything white

the only difference seems to be that it loads something with "xhr" when cosmetic is DISabled

**Comment by okiehsch** [comment]
Well, that is weird, do you disable cosmetic filters via the per site switch?https://github.com/gorhill/uBlock/wiki/Per-site-switches#no-cosmetic-filtering

**Comment by SEXXES** [comment]
no
completely within page settings


but I just tried what you mentioned, same behavior and same results in the log

**Comment by okiehsch** [comment]
> but I just tried what you mentioned, same behavior and same results in the log

Meaning the issue is fixed on our end if you only use the per site switch?

**Comment by SEXXES** [comment]
yes

I "solves" the problem for this specific newsletter but then it may cause "unfiltered" behavior on all the rest of the few hundert e-mails i receive every week… so I won't disable cosmetic filter just for ONE e-mail out of hunderts… thats why I reported this issue

**Comment by okiehsch** [comment]
I was not suggesting you do that, I just wanted to clarify the behaviour on your end :)

If you enable cosmetic filters and add
`@@||protonmail.com^$generichide`
to your filter list this results in ???

**Comment by okiehsch** [comment]
Done.

**Comment by okiehsch** [comment]
Well, that did not work.
![image](https://user-images.githubusercontent.com/16838044/45258851-28899c80-b3c0-11e8-9124-1e9143454973.png)

**Comment by SEXXES** [comment]
before that i forwarded the newsletter to you… not in your mailbox?

**Comment by SEXXES** [comment]
LOL by doing this, I just found 2 new bugs… 😂
by the way I tried it on Firefox Nightly, same behavior there

shame on wordpress and shame on abine.com
you could actually disable my newsletter sign-up by clicking on the link in the forwarded e-mail and block on abine.com, the things that shouldn't be possible!😡

**Comment by okiehsch** [comment]
Ok I got the newsletter via your latest mail.

With a default setup this is what I see
![image](https://user-images.githubusercontent.com/16838044/45258949-5f60b200-b3c2-11e8-8ca7-448e47ee614b.png)

**Comment by SEXXES** [comment]
better then what I've got, but still no pictures shown… I already reported *this issue (not showing pictures) in another bug-report

however, I still don't have an explanation for this behavior, specially because its only that specific newsletter out of hundreds*

**Comment by okiehsch** [comment]
The picture issue is because of the setup of protonmail to not load remote content by default.
![image](https://user-images.githubusercontent.com/16838044/45259029-c0d55080-b3c3-11e8-91e5-8116823b348b.png)

If you allow it

![image](https://user-images.githubusercontent.com/16838044/45259038-ef532b80-b3c3-11e8-9c68-40988217abbb.png)

**Comment by SEXXES** [comment]
yes, but I allowed it too, even "remote content", so thats not the troublemaker in my case

I'll try tomorrow with Safari (and some modifications), I'll let you know as soon as I find out more

**Comment by okiehsch** [comment]
If I understand your issue correctly you can't see the newsletter at all unless you disable cosmetic filters.

This I can't reproduce with a default setup using Chrome or Firefox.
The only difference between your setup and mine seems to be the OS.
I do not have any system running` MacOS`, the issue does not occur on my end using Linux or Windows.

**Comment by SEXXES** [comment]
possible but I still can be lot of other sources… firewall, other add-ons, browser settings, ip/geo location(s)

it's just: normally when I have troubles, it's not just ONE e-mail/newsletter but a lot or even all of them

**Comment by okiehsch** [comment]
Yes, with Fanboys Annoyance List enabled I can reproduce.

`protonmail.com#@#.subscribe-wrapper` will fix your issue.
The filter will not be included in uÀssets, because `Fanboys Annoyance List` is not part of the default setup.

> but that doesn't explain why I don't have the yellow line with Chrome

This bug has been fixed in the latest dev versions, [uBlock Origin development build
1.16.21.5](https://chrome.google.com/webstore/detail/ublock-origin-development/cgbcahbpdhpcegmbfconppldiemgcoii)

**Comment by SEXXES** [comment]
thanks a lot!

**Comment by SEXXES** [comment]
where should I post newsletter fly-ins?
![newsletter fly in](https://user-images.githubusercontent.com/42209961/45269590-1f1c3500-b491-11e8-9f0b-98abbd7b681d.jpeg)

**Comment by okiehsch** [comment]
Certainly not here,
````
If an issue can be fixed without using extended syntax, i.e. script:inject, redirect or procedural cosmetic filters the issue will, as a rule of thumb, not be fixed by uAssets, because fixing it in EasyList if possible is preferable.

Issues involving

anti-adblock
ad-reinsertion
popups/popunders
site breakage
will always be fixed in uAssets even if they don't require extended syntax.

If the ABP-compatible fixes are for high-traffic sites, they should be added to uBO filters until they are added to EasyList.
````

I would say try `Fanboys Annoyance List` https://github.com/ryanbr/fanboy-adblock/issues
